### PR TITLE
SC-5345 - edusharing-content

### PR DESCRIPTION
**Pull request by mwolowyk-cap**
# Description

In this PR we filter out the edusharing results which don't have an external URL since currently the whole content which should be provided is collected from external data sources by the crawler and edusharing contains only the meta data.

## Links to Tickets or other pull requests

https://ticketsystem.schul-cloud.org/browse/SC-5345
https://github.com/schul-cloud/schulcloud-server/pull/1502

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: src/components/organisms/AddContentButton.vue [hunk 1]
```diff
@@ -66,7 +73,7 @@ export default {
 			return response && response.status === 201;
 		},
 		getUrl() {
-			return `/content/${this.resource.ref.id}`;
+			return getMetadataAttribute(this.resource.properties, "cclom:location");
```
Comment (mwolowyk-cap) [inline src/components/organisms/AddContentButton.vue:76]: Konstantin mentioned that result should show directly to resource and not to the DetailPage anymore

File: src/components/organisms/ContentCard.vue [hunk 2]
```diff
@@ -108,12 +108,7 @@ export default {
 			return provider ? provider.replace("/n", "").trim() : "Schul-Cloud";
 		},
 		thumbnail() {
-			return (
-				this.getMetadataAttribute(
-					this.resource.properties,
-					"ccm:thumbnailurl"
-				) || this.resource.preview.url
-			);
+			return this.resource.preview.url;
```
Comment (mwolowyk-cap) [inline src/components/organisms/ContentCard.vue:111]: Image should be loaded from Edusharing location with the redirect to the source due to the CORS

File: src/components/organisms/AddContentButton.vue (hunk 1 above)
Comment (BlackGhostHD) [reply src/components/organisms/AddContentButton.vue:76]: please see https://github.com/schul-cloud/nuxt-client/pull/1116 and request changes if necessary

File: src/components/organisms/ContentCard.unit.js [hunk 3]
```diff
@@ -19,7 +19,7 @@ describe("@components/organisms/ContentCard", () => {
 	it("Renders contentCard img", () => {
 		expect(wrapper.find(".content__img-thumbnail").exists()).toBe(true);
 		expect(wrapper.find(".content__img-thumbnail").attributes("src")).toBe(
-			"https://img.youtube.com/vi/wRBPzExOH2A/maxresdefault.jpg"
+			"https://mv-repo.schul-cloud.org/edu-sharing/preview?nodeId=70eedbd1-a4e9-47ee-ac00-3848863e5295&amp;storeProtocol=workspace&amp;storeId=SpacesStore&amp;dontcache=1593695489846"
```
Comment (BlackGhostHD) [inline src/components/organisms/ContentCard.unit.js:22]: Please write this in the mockData "Resource" and use it as refrence here

File: src/components/organisms/LernstoreDetailView.unit.js [hunk 4]
```diff
@@ -21,7 +21,7 @@ describe("@components/molecules/LernstoreDetailView", () => {
 		expect(wrapper.find(".preview").exists()).toBe(true);
 		expect(wrapper.find(".preview-img").exists()).toBe(true);
 		expect(wrapper.find(".preview-img").attributes("src")).toBe(
-			"https://img.youtube.com/vi/wRBPzExOH2A/maxresdefault.jpg"
+			"https://mv-repo.schul-cloud.org/edu-sharing/preview?nodeId=70eedbd1-a4e9-47ee-ac00-3848863e5295&amp;storeProtocol=workspace&amp;storeId=SpacesStore&amp;dontcache=1593695489846"
```
Comment (BlackGhostHD) [inline src/components/organisms/LernstoreDetailView.unit.js:24]: same here

**Comment by janrenz** [comment]
This could be wrapped by a feature toggle maybe?

File: src/components/organisms/AddContentButton.vue (hunk 1 above)
Comment (mwolowyk-cap) [reply src/components/organisms/AddContentButton.vue:76]: @BlackGhostHD Yes I saw your PR. I will remove this functionality from this one in order not to mix the features

**Comment by mwolowyk-cap** [comment]
> This could be wrapped by a feature toggle maybe?

@janrenz what exactly should be wrapped as feature toggle? Could you please explain.

**Comment by janrenz** [comment]
Just in case we want to disable the filter

**Comment by mwolowyk-cap** [comment]
> Just in case we want to disable the filter

Ok. The filter is in the Server. Then I will add a feature toggle there. See: https://github.com/schul-cloud/schulcloud-server/pull/1502

**Comment by mwolowyk-cap** [comment]
> Just in case we want to disable the filter

Filter is not necessary anymore since we are using guest account, which can access only public content.

File: nuxt.config.js [hunk 5]
```diff
@@ -36,6 +36,7 @@ module.exports = {
 			process.env.JWT_SHOW_TIMEOUT_WARNING_SECONDS,
 		JWT_TIMEOUT_SECONDS: process.env.JWT_TIMEOUT_SECONDS,
 		SC_THEME: process.env.SC_THEME,
+		MODE_LERNSTORE: process.env.LERNSTORE,
```
Comment (ABialoskorski) [inline nuxt.config.js:39]: shouldn't it be the same name?

Comment (mwolowyk-cap) [reply nuxt.config.js:39]: yes. thanks

**Comment by mwolowyk-cap** [comment]
npm install

**Comment by ABialoskorski** [comment]
@schul-cloud-bot update screenshots

**Comment by mwolowyk-cap** [comment]
@schul-cloud-bot update screenshots

**Pull request merged**